core/src/test/java/org/apache/calcite/test/JdbcFrontJdbcBackLinqMiddleTest.java

```diff
@@ -270,6 +270,6 @@
         .query(
             "select \"store\".\"store_country\" as \"c0\", sum(\"inventory_fact_1997\".\"supply_time\") as \"m0\" from \"store\" as \"store\", \"inventory_fact_1997\" as \"inventory_fact_1997\" where \"inventory_fact_1997\".\"store_id\" = \"store\".\"store_id\" group by \"store\".\"store_country\"")
         .planContains(
-            " left.hashJoin(right, new org.apache.calcite.linq4j.function.Function1() {\n");
+            "nestedLoopJoin(left, right, new org.apache.calcite.linq4j.function.Predicate2() {\n");
```

Comment:
The hash join alternative is generated, but finally the nestloop join is chosen by planner.